src/condor_daemon_core.V6/condor_daemon_core.h

```diff
@@ -2347,6 +2361,22 @@ class DaemonCore : public Service
 	void InitSharedPort(bool in_init_dc_command_socket=false);
 
 	std::string m_inherit_parent_sinful;
+
+		// Create a session that permits a remote entity to be an administrator
+		// of this daemon.
+		// - duration: Lifetime of the session.
+		// - capability (output): Resulting serialized session string.
+		// - Return: true on success.
+	bool SetupAdministratorSession(unsigned duration, std::string &capability);
+
+		// Enable remote administration for this daemon.
+	void SetRemoteAdmin(bool remote_admin);
+
+	static unsigned m_remote_admin_seq;
+	static time_t m_startup_time;
```

Comment:
I'd like to see this new startup timestamp tied or unified with the startup timestamp DCCollector::startTime. Probably by setting DCCollector::startTime to match DaemonCore::m_startup_time. Also, change DaemonCore::getStartTime() to use DaemonCore::m_startup_time.